File: TimeTracker.Domain/TimeEntry.cs
using TimeTracker.Domain.Auth;
using TimeTracker.Domain.Base;

namespace TimeTracker.Domain;

public class TimeEntry : BaseModel<Guid>
{
    public DateTimeOffset From { get; set; }
    public DateTimeOffset To { get; set; }
    public string Description { get; set; } = string.Empty;

    public Guid CategoryId { get; set; }
    public Guid UserId { get; set; }
    public Category Category { get; set; } = null!;
    public User User { get; set; } = null!;
}

File: TimeTracker.Frontend.React/src/types/TimeEntry.ts
import type { Category } from "./Category";

export interface TimeEntry {
    id: string;
    description: string;
    from: Date; // ISO date string
    to: Date; // ISO date string
    category: Category | null;
}

export const createBlankTimeEntry = (): TimeEntry => ({
    id: '',
    description: '',
    from: new Date(),
    to: new Date(),
    category: null
});